quantumblacklabs/kedro-viz
# [KED-1423] Feature/metadata panel

## Description

Adds a metadata side panel. 

<img width="937" alt="Screenshot 2020-10-29 at 16 41 18" src="https://user-images.githubusercontent.com/45365769/97604639-a0ced400-1a05-11eb-945a-90fe0c82d9a8.png">

## Development notes

This first version is behind flag `meta` and only shows existing data, a later update will add data from further APIs.

The responsive approach has been adapted to account for the additional sidebar.

## QA notes

The metadata flag must be enabled before testing. To see inputs and outputs, the new graph flag must also be enabled (at least for the initial implementation). Here is an example:

`http://localhost:4141/?data=demo&newgraph=true&meta=true`

Then to see the metadata panel, you should click on any node either in the sidebar or on the chart.

Note that different node types show different information, so test each kind.

To test the responsive approach, test with each sidebar individually and with both open, at sizes below `800px`.

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
